spatialmath/mesh.go

```diff
@@ -27,21 +34,34 @@ type Mesh struct {
 	rawBytes []byte
 }
 
-// The set of supported mesh file types.
-type meshType string
-
-const plyType = meshType("ply")
+// NewMesh creates a mesh from the given triangles and pose.
+// A Mesh created this way should not be attempted to be converted to protobuf
+// as there are not conversion functions to support it currently.
+func NewMesh(pose Pose, triangles []*Triangle, label string) *Mesh {
+	return &Mesh{
+		pose:      pose,
+		triangles: triangles,
+		label:     label,
+	}
+}
 
-func newMeshFromProto(pose Pose, m *commonpb.Mesh, label string) (*Mesh, error) {
-	switch m.FileType {
-	case string(plyType):
-		return NewMeshFromPLY(pose, m.Mesh, label)
-	default:
-		return nil, fmt.Errorf("unsupported Mesh type: %s", m.FileType)
+// NewMeshFromPLYFile is a helper function to create a Mesh geometry from a PLY file.
+func NewMeshFromPLYFile(path string) (*Mesh, error) {
+	//nolint:gosec
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	// nolint:errcheck
+	defer file.Close()
+	bytes, err := io.ReadAll(file)
```

Comment:
My understanding is that ply files may be either binary or ascii, and we make use of both. Will this handle both types of files?

Reply:
Looking at the ply spec it looks like the header is always ascii, and it's only the numbers that are ascii or binary. Writing the parser ourselves should not be very involved, I agree.

I also agree that for right now that would be more than we would want to do, considering how we would have to fork the whole parser.

Can we add a comment to this clarifying this? And also if possible it would be nice to pre-empt that panic by just returning an error, rather than panicking (or capture the panic and return as an error).

Reply:
Ok good idea I'll capture the panic